pkg/cron_workflow.go

```diff
@@ -660,11 +645,75 @@ func (c *Client) GetCronWorkflowStatisticsForTemplates(workflowTemplates ...*Wor
 }
 
 func cronWorkflowColumns(extraColumns ...string) []string {
-	results := []string{"cw.id", "cw.created_at", "cw.uid", "cw.name", "cw.workflow_template_version_id", "cw.manifest"}
+	results := []string{"cw.id", "cw.created_at", "cw.uid", "cw.name", "cw.workflow_template_version_id", "cw.manifest", "cw.namespace"}
 
 	for _, str := range extraColumns {
 		results = append(results, str)
 	}
 
 	return results
 }
+
+func (c *Client) SelectCronWorkflowWithExtraColumns(namespace, uid string, extraColumns ...string) (error, *CronWorkflow) {
+	query, args, err := sb.Select(cronWorkflowColumns(extraColumns...)...).
+		From("cron_workflows cw").
+		Join("workflow_template_versions wtv ON wtv.id = cw.workflow_template_version_id").
+		Join("workflow_templates wt ON wt.id = wtv.workflow_template_id").
+		Where(sq.Eq{
+			"wt.namespace": namespace,
+			"cw.name":      uid,
+		}).
+		ToSql()
+
+	if err != nil {
+		return err, nil
+	}
+
+	cronWorkflow := &CronWorkflow{}
+	if err = c.DB.Get(cronWorkflow, query, args...); err != nil {
+		return err, nil
+	}
+
+	return nil, cronWorkflow
+}
+
+func (c *Client) DeleteCronWorkflowDb(namespace, uid string) error {
+	query, args, err := sb.Select("id").From("cron_workflows").
+		Where(sq.Eq{
+			"uid":       uid,
+			"namespace": namespace,
+		}).ToSql()
+
+	if err != nil {
+		return err
+	}
+	cronWf := &CronWorkflow{}
+	err = c.DB.Get(cronWf, query, args...)
+	if err != nil {
+		return err
+	}
+
+	query, args, err = sb.Delete("cron_workflows").Where(sq.Eq{
+		"id": cronWf.ID,
+	}).ToSql()
+
+	if err != nil {
+		return err
+	}
+
+	if _, err := c.DB.Exec(query, args...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Client) DeleteCronWorkflow(namespace, uid string) error {
```

Comment:
What do you think about renaming this to `DeleteArgoCronWorkflow`?

Reply:
I think this is outdated, method is gone